packages/vue/src/create-element.js

```diff
@@ -1,61 +1,71 @@
+import h from 'hastscript'
+import toH from 'hast-to-hyperscript'
+
 /**
  * MDX default components
  */
-const DEFAULTS = {
+const defaults = {
   inlineCode: 'code',
-  wrapper: 'div'
+  wrapper: {
+    name: 'MDXWrapper',
+    render: function (h) {
+      const children = this.$slots.default
+      return children.length === 1 ? children : h('div', {}, children)
+    }
+  }
 }
 
-/**
- * Renders final tag/component
- * @param {Vue.Component|String} type Element or tag to render
- * @param {Object|Array} props Props and attributes for element
- * @param {Array} children Array of child nodes for component
- * @returns {Vue.VNode} VNode of final rendered element
- */
-export default function(type, props, children) {
-
-  const h = this.createElement
-  const components = this.components
-  const defaults = Object.keys(DEFAULTS)
-
-  let tag
-  let elProps = props
-
-  // We check context to see if the element/tag
-  // is provided in the MDXProvider context.
-  if (Object.keys(components).includes(type)) {
-    // We check to see if props is of type object.
-    // If it is, then we pass them into the MDXContext component
-    const componentProps = typeof props === 'object' ? props : undefined
-    tag = components[type](componentProps)
-
-  } else if (defaults.includes(type)) {
-
-    tag = DEFAULTS[type]
-    // Remove components object from attrs
-    const { components, ...attrs } = elProps.attrs
-    elProps = {
-      attrs
-    }
+const own = {}.hasOwnProperty
+
+export default function (type, props, ...children) {
+  let node
+
+  if (own.call(this.components, type)) {
+    type = this.components[type]
+  } else if (own.call(defaults, type)) {
+    type = defaults[type]
+  }
 
-  // Render final tag if component is not provided in context
+  if (props && typeof props === 'object' && !Array.isArray(props)) {
+    // Empty.
   } else {
-    tag = type
-
-    if (['a', 'input', 'img'].includes(tag)) {
-      const { attrs, ...domProps } = elProps
-      const data = {
-        attrs: attrs,
-        domProps
-      }
-
-      elProps = {
-        ...elProps,
-        ...data
-      }
-    }
+    children.unshift(props)
+    props = {}
+  }
+
+  children = children
+    .flatMap(d => (d == null ? [] : d))
```

Comment:
this seems a bit surprising.
According to the docs, `children` should be a single level array? (either: https://vuejs.org/v2/guide/render-function.html#createElement-Arguments or https://github.com/syntax-tree/hast#parent)
This implementation indicates a multilevel array.

Reply:
it is: react-jsx and vue-jsx babel transforms work in mysterious ways. In my testing, I saw vue do `h(type, [undefined])`. I decided to handle both a children array (including if for some reason `null` or `undefined` are passed), but also support the more react-y way of `h(type, props, child, chilN)`